tests/syscall/epoll/enc/enc.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <netinet/in.h>
+#include <openenclave/enclave.h>
+#include <openenclave/internal/tests.h>
+#include <sys/epoll.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#include <cerrno>
+#include <cstdio>
+
+static const uint16_t _port = 12347;
+static sockaddr_in _addr;
+static int _epfd;
+static int _sockfd;
+
+extern "C" void set_up()
+{
+    OE_TEST(oe_load_module_host_socket_interface() == OE_OK);
+    OE_TEST(oe_load_module_host_epoll() == OE_OK);
+
+    _addr.sin_family = AF_INET;
+    _addr.sin_port = htons(_port);
+    _addr.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
+
+    // create and bind UDP socket
+    _sockfd = socket(AF_INET, SOCK_DGRAM, 0);
+    OE_TEST(_sockfd >= 0);
+    OE_TEST(
+        bind(_sockfd, reinterpret_cast<sockaddr*>(&_addr), sizeof(_addr)) == 0);
+
+    // create epoll instance
+    _epfd = epoll_create(1);
+    OE_TEST(_epfd >= 0);
+}
+
+extern "C" void tear_down()
+{
+    OE_TEST(close(_epfd) == 0);
+    OE_TEST(close(_sockfd) == 0);
+}
+
+extern "C" void wait_for_events()
+{
+    uint8_t run = 1;
+
+    while (run)
+    {
+        epoll_event event{};
+
+        int n;
+        do
+        {
+            n = epoll_wait(_epfd, &event, 1, -1);
```

Comment:
How do you know that the `epoll_wait` ever returned? I think the test would be more complete if you kept a variable that tracks how many events were triggered in total check it at the end.

Reply:
See changes of my last push: The wait thread must have read all the `action_t::run`s before it can read `action_t::stop` and end.